### PR TITLE
ENH: add slits offset and gap to tab completion

### PR DESCRIPTION
**Pull request by vespos**
## Description
Add tab completion upon scientists' request

## How Has This Been Tested?
Tested in xpp3

## Where Has This Been Documented?
N/A

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on travis
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Comment by ZLLentz** [comment]
@vespos do you have the bandwidth to write a pre-release page for these to complete the checklist? The update here looks good.

**Comment by vespos** [comment]
Yes, I'll do that, thanks!

**Comment by vespos** [comment]
@ZLLentz Just added the doc. Is there anything else needed here?

**Comment by ZLLentz** [comment]
I'll throw on a quick edit to fix the header of the doc file and merge :+1:

File: docs/source/upcoming_release_notes/1033-slit-tab-completion.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+IssueNumber Title
```
Comment (ZLLentz) [inline docs/source/upcoming_release_notes/1033-slit-tab-completion.rst:1]:
```suggestion
1033 Slit Tab Completion
```

**Review by ZLLentz — APPROVED** [review]

**Pull request merged**